base/math.jl

```diff
@@ -582,6 +582,10 @@ function _hypot(x::NTuple{N,<:Number}) where {N}
 		return r
 	end
 end
+# FIXME: revisit when #30387 is resolved
+_floatmin(x::AbstractFloat) = _floatmin(typeof(x))
+_floatmin(::Type{T}) where {T<:AbstractFloat} = nextfloat(zero(T)) / eps(T)
```

Comment:
A slight optimization: `_floatmin(::Type{T}) where {T<:IEEEFloat} = floatmin(T)`.   This gives the same results, but is faster for `Float16`.

Reply:
I'm also not sure how I feel about defining a `_floatmin` just for this function. It seems like we need a better way to figure out "is this thing a float-ey object, but not an `AbstractFloat`"?